### ISSUE TITLE
Host firewall drops replies when first enforcing policies

### ISSUE THREAD

**Issue by pchaigno** [cilium/cilium#25448]
See https://github.com/cilium/cilium/commit/439a0a059fdcabe23a33b427b637494bc5a59eda.

TL;DR. We have an issue with the host firewall where, the first time we enforce policies, we may drop reply traffic for legitimate connection that should be allowed by the policies. The workaround we implemented in CI is to enable, disable, and reenable host policy enforcement. Since the issue only happen on the first enforcement, that "fixes" it.

This can however impact users of the feature to some extent and it's not clear how to improve thing. It's unclear that we can even avoid this issue.

In any case, we should at least document it since it is known to cause issues (see e.g. CI flake https://github.com/cilium/cilium/issues/15455).

**Comment by NeonSludge** [comment]
I believe we're seeing this exact issue in our cluster (1.13.4, strict KPR, BPF masquerading).

What happens is when we first apply CCNPs with node selectors in a fresh Kubernetes cluster it becomes unusable for a while because kube-apiserver and etcd connections get disrupted (we have, of course, added ingress rules for them). The situation seems to resolve on its own after about 5 minutes, the cluster comes back up and any changes to CCNPs after that time never lead to any networking issues.

**Comment by hahahaheihei** [comment]
I had the same problem, 
kubernetes version 1.21
cilium version v1.15.6 
i deployed a CiliumClusterwideNetworkPolicy to implement white list network control ,when i restart cilium-agent ,node network in the whitelist are dropped
